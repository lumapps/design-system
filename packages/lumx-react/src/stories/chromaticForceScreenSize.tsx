/* eslint-disable */
import React from 'react';
import isChromatic from 'chromatic/isChromatic';

/** Story decorator used to force a minimum screen size when running in chromatic */
export const chromaticForceScreenSize = (story: any) =>
    isChromatic() ? <div style={{ minWidth: 1200, minHeight: 800 }}>{story()}</div> : story();
