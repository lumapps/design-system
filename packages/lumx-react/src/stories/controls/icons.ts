import {
    mdiAbTesting,
    mdiAbjadArabic,
    mdiAccount,
    mdiAccountBox,
    mdiAlert,
    mdiAlertCircle,
    mdiArrowDown,
    mdiArrowUp,
    mdiAtom,
    mdiBee,
    mdiBell,
    mdiBullhornOutline,
    mdiCheck,
    mdiCheckCircle,
    mdiChevronDown,
    mdiChevronLeft,
    mdiChevronRight,
    mdiChevronUp,
    mdiClose,
    mdiCloseCircle,
    mdiDelete,
    mdiDotsHorizontal,
    mdiDragVertical,
    mdiEarth,
    mdiEmail,
    mdiEye,
    mdiFileEdit,
    mdiFlag,
    mdiFolder,
    mdiFolderGoogleDrive,
    mdiFoodApple,
    mdiGoogleCirclesExtended,
    mdiHeart,
    mdiHome,
    mdiImageBroken,
    mdiInformation,
    mdiLink,
    mdiMagnifyMinusOutline,
    mdiMagnifyPlusOutline,
    mdiMenuDown,
    mdiMessageTextOutline,
    mdiMinus,
    mdiOpenInNew,
    mdiPauseCircleOutline,
    mdiPencil,
    mdiPlay,
    mdiPlayCircleOutline,
    mdiPlus,
    mdiRadioboxBlank,
    mdiRadioboxMarked,
    mdiReply,
    mdiSend,
    mdiStar,
    mdiTextBox,
    mdiTextBoxPlus,
    mdiTram,
    mdiTranslate,
    mdiViewList,
} from '@lumx/icons';

// Small selection of mdi icons to
export const iconArgType = {
    control: { type: 'select' },
    options: {
        mdiAbTesting,
        mdiAbjadArabic,
        mdiAccount,
        mdiAccountBox,
        mdiAlert,
        mdiAlertCircle,
        mdiArrowDown,
        mdiArrowUp,
        mdiAtom,
        mdiBee,
        mdiBell,
        mdiBullhornOutline,
        mdiCheck,
        mdiCheckCircle,
        mdiChevronDown,
        mdiChevronLeft,
        mdiChevronRight,
        mdiChevronUp,
        mdiClose,
        mdiCloseCircle,
        mdiDelete,
        mdiDotsHorizontal,
        mdiDragVertical,
        mdiEarth,
        mdiEmail,
        mdiEye,
        mdiFileEdit,
        mdiFlag,
        mdiFolder,
        mdiFolderGoogleDrive,
        mdiFoodApple,
        mdiGoogleCirclesExtended,
        mdiHeart,
        mdiHome,
        mdiImageBroken,
        mdiInformation,
        mdiLink,
        mdiMagnifyMinusOutline,
        mdiMagnifyPlusOutline,
        mdiMenuDown,
        mdiMessageTextOutline,
        mdiMinus,
        mdiOpenInNew,
        mdiPauseCircleOutline,
        mdiPencil,
        mdiPlay,
        mdiPlayCircleOutline,
        mdiPlus,
        mdiRadioboxBlank,
        mdiRadioboxMarked,
        mdiReply,
        mdiSend,
        mdiStar,
        mdiTextBox,
        mdiTextBoxPlus,
        mdiTram,
        mdiTranslate,
        mdiViewList,
    },
};
