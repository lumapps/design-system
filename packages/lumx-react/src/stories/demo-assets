../../../site-demo/static/demo-assets