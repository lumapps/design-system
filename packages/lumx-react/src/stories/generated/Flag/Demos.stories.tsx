/**
 * File generated when storybook is started. Do not edit directly!
 */
export default { title: 'LumX components/flag/Flag Demos' };

export { App as Flag } from './flag';
