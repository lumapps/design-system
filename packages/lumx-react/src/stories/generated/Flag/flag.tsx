../../../../../site-demo/content/product/components/flag/react/flag.tsx