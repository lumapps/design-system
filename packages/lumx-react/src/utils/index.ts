/**
 * WARNING: All modules exported here are exposed to NPM in '@lumx/react/utils'.
 */

export { ClickAwayProvider } from './ClickAwayProvider';
export { Portal, type PortalProps, type PortalInit, PortalProvider, type PortalProviderProps } from './Portal';
