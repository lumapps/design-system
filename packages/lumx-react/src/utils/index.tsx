export * from '@lumx/core/js/utils';

export * from './getRootClassName';

export * from './htmlDecode';

export * from './partitionMulti';

export * from './type';
