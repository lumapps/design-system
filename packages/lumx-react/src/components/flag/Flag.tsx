import React, { forwardRef } from 'react';
import classNames from 'classnames';

import { ColorPalette, ColorVariant, Icon, Size, Theme } from '@lumx/react';
import { Comp, GenericProps, getRootClassName, handleBasicClasses } from '@lumx/react/utils';

export interface FlagProps extends GenericProps {
    /** Theme adapting the component to light or dark background. */
    theme?: Theme;
    /** Color of the component. */
    color?: ColorPalette;
    /** Icon to use before the label. */
    icon?: string;
    /** Text label of the flag. */
    label: string;
}

const COMPONENT_NAME = 'Flag';
const CLASSNAME = getRootClassName(COMPONENT_NAME);
const DEFAULT_PROPS: Partial<FlagProps> = {
    theme: Theme.light,
};

/**
 * Flag component.
 *
 * @param  props Component props.
 * @param  ref   Component ref.
 * @return React element.
 */
export const Flag: Comp<FlagProps, HTMLDivElement> = forwardRef((props, ref) => {
    const { label, icon, color, className, theme, ...forwardedProps } = props;
    const flagColor = color || (theme === Theme.light ? ColorPalette.dark : ColorPalette.light);

    return (
        <div
            {...forwardedProps}
            className={classNames(className, handleBasicClasses({ prefix: CLASSNAME, color: flagColor }))}
            ref={ref}
        >
            {icon && (
                <Icon
                    icon={icon}
                    color={color}
                    colorVariant={ColorVariant.D2}
                    size={Size.xxs}
                    className={`${CLASSNAME}__icon`}
                />
            )}
            <span className={`${CLASSNAME}__label`}>{label}</span>
        </div>
    );
});
Flag.displayName = COMPONENT_NAME;
Flag.className = CLASSNAME;
Flag.defaultProps = DEFAULT_PROPS;
