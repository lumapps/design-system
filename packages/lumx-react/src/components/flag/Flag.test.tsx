import { mount, shallow } from 'enzyme';
import 'jest-enzyme';
import React, { ReactElement } from 'react';

import { ColorPalette, Theme } from '@lumx/react';
import { mdiAbTesting } from '@lumx/icons';
import { itShouldRenderStories, commonTestsSuite, Wrapper } from '@lumx/react/testing/utils';
import { getBasicClass } from '@lumx/react/utils';

import { Flag, FlagProps } from './Flag';
import * as stories from './Flag.stories';

const CLASSNAME = Flag.className as string;

/**
 * Mounts the component and returns common DOM elements / data needed in multiple tests further down.
 */
const setup = (propOverrides: Partial<FlagProps> = {}, shallowRendering = true) => {
    const props = {
        label: 'default',
        ...propOverrides,
    };
    const renderer: (el: ReactElement) => Wrapper = shallowRendering ? shallow : mount;
    const wrapper = renderer(<Flag {...props} />);

    return {
        props,
        iconEl: wrapper.find('Icon'),
        wrapper,
    };
};

describe(`<${Flag.displayName} />`, () => {
    // 1. Test render via snapshot (default state of component).
    describe('Snapshots and structure', () => {
        itShouldRenderStories(stories, Flag);
    });

    // 4. Test conditions (i.e. things that display or not in the UI based on props).
    describe('Conditions', () => {
        it('should use the icon', () => {
            const { iconEl } = setup({ icon: mdiAbTesting });

            expect(iconEl).toBeDefined();
            expect(iconEl.prop('icon')).toEqual(mdiAbTesting);
        });

        it('should have correct default color', () => {
            const { wrapper } = setup({});
            expect(wrapper).toHaveClassName(
                getBasicClass({
                    prefix: CLASSNAME,
                    type: 'color',
                    value: ColorPalette.dark,
                }),
            );
        });

        it('should switch color with theme', () => {
            const { wrapper } = setup({ theme: Theme.dark });
            expect(wrapper).toHaveClassName(
                getBasicClass({
                    prefix: CLASSNAME,
                    type: 'color',
                    value: ColorPalette.light,
                }),
            );
        });

        it('should use the color', () => {
            const color = ColorPalette.green;
            const { wrapper, iconEl } = setup({ icon: mdiAbTesting, color });

            expect(iconEl.prop('color')).toEqual(color);
            expect(wrapper).toHaveClassName(
                getBasicClass({
                    prefix: CLASSNAME,
                    type: 'color',
                    value: color,
                }),
            );
        });
    });

    // Common tests suite.
    commonTestsSuite(setup, { className: 'wrapper', prop: 'wrapper' }, { className: CLASSNAME });
});
