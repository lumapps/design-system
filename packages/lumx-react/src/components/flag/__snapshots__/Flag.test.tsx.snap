// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flag /> Snapshots and structure should render story 'defaultProps' 1`] = `
<div
  className="lumx-flag lumx-flag--color-dark"
>
  <span
    className="lumx-flag__label"
  >
    hearing
  </span>
</div>
`;

exports[`<Flag /> Snapshots and structure should render story 'withColor' 1`] = `
Array [
  <div
    className="lumx-flag lumx-flag--color-blue"
  >
    <Icon
      className="lumx-flag__icon"
      color="blue"
      colorVariant="D2"
      icon="M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C9.24,3 10.91,3.81 12,5.08C13.09,3.81 14.76,3 16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35Z"
      size="xxs"
    />
    <span
      className="lumx-flag__label"
    >
      blue
    </span>
  </div>,
  <div
    className="lumx-flag lumx-flag--color-dark"
  >
    <Icon
      className="lumx-flag__icon"
      color="dark"
      colorVariant="D2"
      icon="M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C9.24,3 10.91,3.81 12,5.08C13.09,3.81 14.76,3 16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35Z"
      size="xxs"
    />
    <span
      className="lumx-flag__label"
    >
      dark
    </span>
  </div>,
  <div
    className="lumx-flag lumx-flag--color-green"
  >
    <Icon
      className="lumx-flag__icon"
      color="green"
      colorVariant="D2"
      icon="M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C9.24,3 10.91,3.81 12,5.08C13.09,3.81 14.76,3 16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35Z"
      size="xxs"
    />
    <span
      className="lumx-flag__label"
    >
      green
    </span>
  </div>,
  <div
    className="lumx-flag lumx-flag--color-primary"
  >
    <Icon
      className="lumx-flag__icon"
      color="primary"
      colorVariant="D2"
      icon="M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C9.24,3 10.91,3.81 12,5.08C13.09,3.81 14.76,3 16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35Z"
      size="xxs"
    />
    <span
      className="lumx-flag__label"
    >
      primary
    </span>
  </div>,
  <div
    className="lumx-flag lumx-flag--color-red"
  >
    <Icon
      className="lumx-flag__icon"
      color="red"
      colorVariant="D2"
      icon="M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C9.24,3 10.91,3.81 12,5.08C13.09,3.81 14.76,3 16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35Z"
      size="xxs"
    />
    <span
      className="lumx-flag__label"
    >
      red
    </span>
  </div>,
  <div
    className="lumx-flag lumx-flag--color-secondary"
  >
    <Icon
      className="lumx-flag__icon"
      color="secondary"
      colorVariant="D2"
      icon="M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C9.24,3 10.91,3.81 12,5.08C13.09,3.81 14.76,3 16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35Z"
      size="xxs"
    />
    <span
      className="lumx-flag__label"
    >
      secondary
    </span>
  </div>,
  <div
    className="lumx-flag lumx-flag--color-yellow"
  >
    <Icon
      className="lumx-flag__icon"
      color="yellow"
      colorVariant="D2"
      icon="M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C9.24,3 10.91,3.81 12,5.08C13.09,3.81 14.76,3 16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35Z"
      size="xxs"
    />
    <span
      className="lumx-flag__label"
    >
      yellow
    </span>
  </div>,
]
`;

exports[`<Flag /> Snapshots and structure should render story 'withIcon' 1`] = `
<div
  className="lumx-flag lumx-flag--color-dark"
>
  <Icon
    className="lumx-flag__icon"
    colorVariant="D2"
    icon="M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C9.24,3 10.91,3.81 12,5.08C13.09,3.81 14.76,3 16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35Z"
    size="xxs"
  />
  <span
    className="lumx-flag__label"
  >
    hearing
  </span>
</div>
`;
